src/utils/validate-handler-config.ts

```diff
@@ -1,10 +1,44 @@
 import { CommandHandlerConfig } from '../types/CommandHandlerConfig.js'
 import { EventHandlerConfig } from '../types/EventHandlerConfig.js'
 
+// Validate config to be appropriate value
 export const validateEventHandlerConfig = (
   config: unknown,
-): config is EventHandlerConfig => true // TODO: Validate config to be appropriate value
+): config is EventHandlerConfig => {
+  if (typeof config !== 'object' || config === null) {
+    throw new TypeError('Config must be an object')
+  }
+  if (!('eventName' in config) || typeof config.eventName !== 'string')
+    throw new TypeError('Config must have eventName property of type string')
 
+  // once is optional
+  if ('once' in config && typeof config.once !== 'boolean')
+    throw new TypeError('once must be of type boolean if it exists in config')
+
+  if (!('execute' in config) || typeof config.execute !== 'function')
+    throw new TypeError('Config must have execute property of type function')
+
```

Comment:
We already have [zod](https://zod.dev/) installed for validating things. Can you use that instead? Thank you 🙏